packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -594,6 +594,29 @@ test.describe('Invalidation', () => {
 		await page.locator('button').click();
 		await expect(page.getByText('updated')).toBeVisible();
 	});
+
+	test('goto after invalidation does not reset state', async ({ page }) => {
+		await page.goto('/load/invalidation/invalidate-then-goto');
+		const layout = await page.textContent('p.layout');
+		const _page = await page.textContent('p.page');
+		expect(layout).toBeDefined();
+		expect(_page).toBeDefined();
+
+		await page.click('button.invalidate');
+		await page.evaluate(() => window.promise);
+		const next_layout_1 = await page.textContent('p.layout');
+		const next_page_1 = await page.textContent('p.page');
+		expect(next_layout_1).not.toBe(layout);
+		expect(next_page_1).toBe(_page);
+
+		await page.click('button.goto');
+		await page.evaluate(() => window.promise);
+		const next_layout_2 = await page.textContent('p.layout');
+		const next_page_2 = await page.textContent('p.page');
+		expect(next_layout_2).toBe(next_layout_1);
+		expect(next_layout_2).not.toBe(layout);
+		expect(next_page_2).not.toBe(next_page_1);
```

Comment:
you could write it like this:
```suggestion
		await page.click('button.invalidate');
		expect(page.locator('p.layout').textContent()).not.toBe(layout);
		expect(page.locator('p.page').textContent()).toBe(_page);

		const layout2 = await page.textContent('p.layout');
		const page2 = await page.textContent('p.page');
		await page.click('button.goto');
		expect(page.locator('p.page')).not.toBe(page2);
		expect(page.locator('p.layout')).toBe(layout2);
```

Reply:
I don't know, it reads weird to me (first compare text content, then save that same selector to a variable). It also is missing one `expect` check at the end.
Also the present test mimicks the other tests in that suite, and frankly if this is the only thing holding up this PR then I'd rather discuss this separately and merge this PR because it's an important bug to fix.

Reply:
If we do leave it as is then we were going to rename `window.promise` to something more unique, right?